site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -1077,6 +1097,24 @@ private function updateGradeable(Gradeable $gradeable, $details) {
                 continue;
             }
 
+            if (in_array($prop, $depend_props)) {
+                if ($post_val === "") {
+                    $post_val = null;
+                }
+            }
+
+            if ($prop === "depends_on") {
+                try {
+                    $gradeables = new GradeableList($this->core);
```

Comment:
Why are you constructing a whole GradeableList here and not just doing a simple existence query check?